experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -643,35 +619,15 @@ function compareCellPositions(
 	baseMark: Mark<unknown>,
 	newMark: EmptyInputCellMark<unknown>,
 	newIntention: RevisionTag | undefined,
-	cancelledInserts: Set<RevisionTag>,
 	metadata: RevisionMetadataSource,
 ): number {
 	const newCellId = getInputCellId(newMark, newIntention, metadata);
 	assert(newCellId !== undefined, 0x71f /* Should have cell ID */);
-	if (baseCellId.revision === newCellId.revision) {
-		if (isNewAttach(newMark)) {
-			// There is some change foo that is being cancelled out as part of a rebase sandwich.
-			// The marks that make up this change (and its inverse) may be broken up differently between the base
-			// changeset and the new changeset because either changeset may have been composed with other changes
-			// whose marks may now be interleaved with the marks that represent foo/its inverse.
-			// This means that the base and new marks may not be of the same length.
-			// We do however know that the all of the marks for foo will appear in the base changeset and all of the
-			// marks for the inverse of foo will appear in the new changeset, so we can be confident that whenever
-			// we encounter such pairs of marks, they do line up such that they describe changes to the same first
-			// cell. This means we can safely treat them as inverses of one another.
-			return 0;
-		}
-
-		if (
-			areOverlappingIdRanges(
-				baseCellId.localId,
-				baseMark.count,
-				newCellId.localId,
-				newMark.count,
-			)
-		) {
-			return baseCellId.localId - newCellId.localId;
-		}
+	if (
+		baseCellId.revision === newCellId.revision &&
+		areOverlappingIdRanges(baseCellId.localId, baseMark.count, newCellId.localId, newMark.count)
```

Comment:
Looks like if they are targetting the same revision, but they do not overlap, when we want to handle it immediately by looking at the `adjacentCells` information. Here's are repo:
```
	it("same revision - no overlap", () => {
		const adjacentCells: SF.IdRange[] = [{ id: brand(1), count: 3 }];
		const cellA = Mark.modify("A", { revision: tag1, localId: brand(1), adjacentCells });
		const cellC = Mark.modify("C", { revision: tag1, localId: brand(3), adjacentCells });

		const composedAC = shallowCompose([tagChange([cellA], tag2), tagChange([cellC], tag3)]);
		const composedCA = shallowCompose([tagChange([cellC], tag3), tagChange([cellA], tag2)]);

		const expected = [cellA, cellC];

		assert.deepEqual(composedAC, expected);
		assert.deepEqual(composedCA, expected);
	});
```

Reply:
Nice catch. We do check this in the rebase version of `compareCellPositions`. I'll see if we can merge this part of the two functions.